legit.py

```diff
@@ -0,0 +1,62 @@
+from git import Repo
+from git.exc import GitCommandError
+from typing import Iterable
+# import logging
+
+# parent_commits: Tuple[str, str] = (), author: Tuple[str, str] = author,
+# committer: Tuple[str, str] = committer
+# author = Actor("An author", "katridi@yandex.ru")
+# committer = Actor("A committer", "katridi@yandex.ru")
+# TODO Consider author and commiter param
+# debug, add logging, write tests
+
+
+class Repository:
+
+    def __init__(self, path: str) -> None:
+        self.repo = Repo.init(path)
+
+    @classmethod
+    def clone(cls, url: str, path: str) -> str:
+        try:
+            Repo.clone_from(url, path)
+        except GitCommandError:
+            return None
+        return cls(path)
+
+    @classmethod
+    def init(cls, path: str) -> str:
+        try:
+            Repo.init(path)
+        except GitCommandError:
+            return None
+        return cls(path)
+
+    def create_branch(self, branch_name: str) -> None:
+        try:
+            self.repo.git.checkout('-b', branch_name)
+        except GitCommandError:
+            self.repo.git.checkout(branch_name)
+
+    def commit(self, message: str) -> None:
+        self.repo.index.commit(message)
+
+    def add_changes(self, files: Iterable[str] = None) -> None:
+        if files:
+            for file in files:
+                self.repo.index.add(file)
+        else:
+            self.repo.git.add(A=True)
+
+    def push_to_origin(self, branch_name: str) -> None:
+        self.repo.head.reset(index=True, working_tree=True)
+        self.repo.git.pull("origin", branch_name)
+        self.repo.git.push("origin", 'HEAD:' + branch_name)
+
+    def merge_branch(self, merge_from: str, merge_into: str, merge_commit_message: str) -> None:
```

Comment:
maybe it should be to move the args in this way: https://gist.github.com/jekoff/70f966180b704699ff77e4c9ac883748

Reply:
so far lenght of this string is 97